app/readappdata.go

```diff
@@ -0,0 +1,45 @@
+package app
+
+import (
+	"bufio"
+	"os"
+	"path/filepath"
+)
+
+func stdinToAppData() ([]byte, error) {
+	return scannerToAppData(bufio.NewScanner(os.Stdin))
+}
+
+func filePathToAppData(filePath string) ([]byte, error) {
+	filePath, err := filepath.Abs(filePath)
+	if err != nil {
+		return nil, err
+	}
+	f, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	return scannerToAppData(bufio.NewScanner(f))
+}
+
+func scannerToAppData(scanner *bufio.Scanner) ([]byte, error) {
```

Comment:
Why are we using `bufio.Scanner` instead of `bufio.Reader`? Perhaps I'm missing something but it doesn't seem like we're reading tokenized input

Reply:
I have no idea, good question, what benefit does Reader give over Scanner?

Reply:
In my mind, I thought Reader was the more "default" option if you want buffered IO whereas Scanner is specifically for scanning chunks of input by delimiter https://pkg.go.dev/bufio#Scanner

If this has been lifted from the existing code, lets defer investigating that to any TD follow-up work. It might be necessary for reading input arguments, which are space delimited.

Reply:
yes it was in the original code.

Lets leave as TD for now and can look up the differences and advantages of each method :)